test/e2e/query_frontend_test.go

```diff
@@ -387,6 +387,27 @@ func TestQueryFrontend(t *testing.T) {
 	})
 }
 
+// https://github.com/thanos-io/thanos/issues/3570
+// "The query splitting test case is flaky because we are using the current timestamp to send queries, which makes the splitting behavior non-deterministic"
+// (Wiard)
+// This should be resolved in a much better way, but for now it removes the flakyness.
+func didItSplitValue(n float64, nightSwitch bool) float64 {
```

Comment:
I don't get the meaning of the variable `nightSwitch`, I feel like it's not descriptive :thinking: or maybe I am stupid :joy: Why for `cortex_cache_*` metrics we don't apply the doubling logic after the first if branch?

Reply:
To be honest I thought I knew it yesterday but as I was trying to describe it here, it did not make any sense. I agree that if it would split, you should expect double the values on every related metric - at every timestamp. Now we only have that behavior when the hour is '1', and it's different on hour '0'. 

What I'm now trying to achieve is to actually remove the flakiness of this test. The internals have not changed, we merely hotfixed a test not fail if you try to run the test at a specific time of day. By no means should this be THE fix, it's a temporary thing that requires more effort. 

I've re-opened https://github.com/thanos-io/thanos/issues/3570 and placed my comment there as well. Like, this requires some investigation. If it does make sense, we should create better tests and possibly update the documentation.

I understand if we don't merge this, to create further confusion perhaps.